### PR TITLE
Raymondli9112/acronyms

### PR DESCRIPTION
**Pull request by raymondli9112**
If your PR is related to a contribution to the taxonomy, please, fill

out the following questionnaire. If not, replace this whole text and the

following questionnaire with whatever information is applicable to your PR.

**Describe the contribution to the taxonomy**

<!-- A concise description of what the contribution brings, replace "..." in the bullet list -->

- editing documents to expand acronyms

- shows the user which sentence(s) were changed and what changes were made to those sentence(s)

- gives the user the corrected document, preserving it's original formal. 

**Input given at the prompt**

<!-- What you entered, replace "..." -->

```

Given the following document, correct each sentence that has an acronym so that all acronyms are expanded based on the rules of the style guide. First give the incorrect version of the sentence and then provide the corrected version of that respective sentence. After all sentences with an acronym have been edited, update the entire document and redisplay in its final and correct form, preserving the original format the best you can. If there are no acronyms to be expanded, please output the following statement: "The document looks great :)" 

Style guide:  

1. Do not use abbreviations or acronyms that the reader would not quickly recognize, instead expand the acronym.

2. Do not follow an organization's full name with an abbreviation or acronym in parentheses. 

document: "Electric Vehicles EV have revolutionized the automotive industry"

```

**Response from the original model**

<!-- What you received from the original model in response to your input, 

replace "..." -->

```

 Electric Vehicles (EV) have revolutionized the automotive industry.           

```

**Response from the fine-tuned model**

<!-- Generate a synthetic dataset based on your newly added seed data; train the model 

with the synthetic data and now re-test the model's response with the same prompt.

Replace "..." with what you receive with the finetuned model. -->

```
Style guide:                                                                                                                                                                             
1. Do not use abbreviations or acronyms that the reader would not quickly recognize, instead expand the acronym.                                                                                                                                                                                                                                                       
2. Do not follow an organization's full name with an abbreviation or acronym in parentheses.                                                                                               document: "Electric Vehicles EV have revolutionized the automotive industry"                                                                                                              
corrected version: "Electric vehicles with cutting-edge technology have revolutionized the automotive industry"                                                                           document: "FL is currently ranked 2nd in the nation for EV adoption as more and more Floridians opt for an electric vehicle."                                                             
corrected version: "The state of Florida is currently ranked second in the nation for electric vehicle adoption as more and more residents choose electric vehicles over gas-powered ones."        
```

**Contribution checklist**

<!-- Insert an x between the empty brackets: [ ] >> [x] -->

- [ ] The contribution was tested with `ilab generate`

- [ ] No errors or warnings were produced by `ilab generate`

- [ ] All [commits are signed off](https://github.com/instructlab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)

- [ ] The `qna.yaml` file contains at least 5 `seed_examples`

- [ ] The `qna.yaml` file was [linted](https://yamllint.com)/ and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)

- [ ] An `attribution.txt` file in the same folder as the `qna.yaml` file

- [ ] Content does not include PII or otherwise sensitive or confidential information

- [ ] Content does not include anything documented in the project's [Avoid these Topics](https://github.com/instructlab/taxonomy/blob/main/docs/SKILLS_GUIDE.md#avoid-these-topics) guidelines

### CODE REVIEW

**Review by jjasghar — CHANGES_REQUESTED** [review]
Please fix the linting issue(s) located here: https://github.com/instructlab/taxonomy/pull/1191/files and the DCO.

**Review by jjasghar — CHANGES_REQUESTED** [review]
Please fix the linting issue(s) located here: https://github.com/instructlab/taxonomy/pull/1191/files and DCO too please.

**Comment by raymondli9112** [comment]
I've fixed the linting issues by adding the attribution.txt file

File: compositional_skills/writing/grounded/acronyms/qna.yaml
Comment (jjasghar) [inline compositional_skills/writing/grounded/acronyms/qna.yaml:1]: Please remove this file.

**Review by jjasghar — CHANGES_REQUESTED** [review]
Please remove the extra qna.yaml, and fix the DCO. And if you can squash down to one commit.

File: compositional_skills/writing/grounded/acronyms/qna.yaml
Comment (raymondli9112) [reply compositional_skills/writing/grounded/acronyms/qna.yaml:1]: I've deleted the extra qna.yml file. I'm not sure what to fix in the DCO though.

Comment (jjasghar) [reply compositional_skills/writing/grounded/acronyms/qna.yaml:1]: I fixed it for you.

**Comment by jjasghar** [comment]
@instructlab-bot precheck

**Comment by jjasghar** [comment]
@instructlab-bot generate

**Comment by obuzek** [comment]
I see what you’re trying to do here, but as written this isn’t a great fit for this process.

I’m noticing a couple things from your SDG output:

1. A lot of your generated questions are getting infected by the types of organizations being mentioned in the context.
2. The instructions you’re writing in both the `question`s and the `task_description`s are getting lost - likely they’re coming into conflict with the invisible-to-you system prompt being provided during SDG.

The InstructLab process is more effective when you apply it to sub-parts of skills, rather than trying to generate a complex workflow like this.

I recommend a resubmit wherein you simplify the questions and answers to something like:

```
context: I’m an astronaut for NASA.
question: Rewrite the sentence in the context, expanding all acronyms.
answer: I’m an astronaut for the National Aeronautics and Space Administration.
```

Improving the acronym expanding skill should likely accomplish what you’re trying to do, which is make the ultimate tuned model better at responding to editing prompts.

**Review by jjasghar — CHANGES_REQUESTED** [review]
Please fix the linting issues, then we can move this over to e2e-ready.

**Comment by jjasghar** [comment]
Hi! :wave:
With the new update and release of 0.18.0, our [base knowledge submissions](https://github.com/instructlab/taxonomy/?tab=readme-ov-file#knowledge-yaml-examples) have drastically changed. This has given us an opportunity to significantly improve the accuracy of our fine-tuned models, which is a major win for our community!
But alas, that has caused some changes that this PR isn't able to continue forward. We greatly appreciate the work you've put into this, and wish we could accept it but the amount of work to get it to be `version: 3` is unrealistic for us to ask.
If you _do_ want to do this, please don't hesitate to take the `qna.yaml` update it, and build it out. Otherwise, we hope you come back to our community with a new submission and help build out our Open Source community-driven LLM.

If you have any questions directly, please don't hesitate to comment on the PR or email me at awesomeATinstructlab.ai (yes, really), and I'll do my best to answer your questions.
